spacy/ml/tb_framework.pyx

```diff
@@ -249,9 +249,11 @@ cdef list _parse_batch(CBlas cblas, TransitionSystem moves, StateC** states,
     cdef np.ndarray step_actions
 
     scores = []
-    while sizes.states >= 1:
+    while sizes.states >= 1 and (actions is None or actions):
         step_scores = numpy.empty((sizes.states, sizes.classes), dtype="f")
         step_actions = actions[0] if actions is not None else None
+        assert step_actions is None or step_actions.size == sizes.states, \
+            f"number of step actions ({step_actions.size}) must equal number of states ({sizes.states})"
```

Comment:
Why do you propose an `assert` with a fail message here? If this is user-facing, we'd rather have an exception + custom error message.

Reply:
Discussed yesterday with @svlandeg , both assertions are to catch programming errors.